src/controllers/Document.ts

```diff
@@ -773,6 +773,38 @@ export class DocumentController extends BaseController {
       .then(response => response.result);
   }
 
+  /**
+   * Applies a partial update to an existing document.
+   * If the document doesn't already exist, a new document is created.
+   *
+   * @param index Index name
+   * @param collection Collection name
+   * @param _id Unique document identifier
+   * @param body Partial changes to apply to the document and Fields to add to the document if it gets created (optional)
+   * @param options
+   * @returns {Promise<Object>}
+   * @see https://docs.kuzzle.io/sdk/js/7/controllers/document/upsert/
+  */
+  upsert (
+    index: string,
+    collection: string,
+    _id: string,
+    body: JSONObject,
+    options: {refresh?: string, retryOnConflict?: boolean, source?: boolean} = {}
```

Comment:
`retryOnConflict` is not added to the request. Since this is a common argument, you should check for it's presence in the `Kuzzle.query` method and add it if present (this bug already exists in prior version of the SDK)